cooper/optim/constrained_optimizers/alternating_optimizer.py

```diff
@@ -89,9 +67,9 @@ class AlternatingPrimalDualOptimizer(BaseAlternatingOptimizer):
 
     def roll(
         self,
-        compute_cmp_state_fn: Callable[..., CMPState],
-        compute_violations_fn: Optional[Callable[..., CMPState]] = None,
-    ) -> tuple[CMPState, LagrangianStore]:
+        compute_cmp_state_kwargs: dict = {},
+        compute_violations_kwargs: dict = {},
+    ) -> tuple[CMPState, LagrangianStore, LagrangianStore]:
```

Comment:
Remember to update type hint after returning the loss.
Also, could define the type above so that the signature of roll fits in one/three lines

Reply:
This type idea has been captured by the `RollOut` dataclass implemented in #90